stl/inc/xlocnum

```diff
@@ -1338,8 +1348,8 @@ protected:
     }
 
 private:
-    char* __CLRCALL_OR_CDECL _Ffmt(
-        char* _Fmt, char _Spec, ios_base::fmtflags _Flags) const { // generate sprintf format for floating-point
+    char* __CLRCALL_OR_CDECL _Ffmt(char* const _Fmt, const char _Spec, const ios_base::fmtflags _Flags,
+        const bool _Has_precision) const { // generate sprintf format for floating-point

```

Comment:
This is changing the signature of a DLL-exported member function, which breaks binary compatibility. (The obscure clue is that this is marked with the `__CLRCALL_OR_CDECL` calling convention; we mark separately compiled functions with explicit calling conventions.) Here's the export:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29115\bin\Hostx64\x64>dumpbin /exports msvcp140.dll | findstr _Ffmt
        424  1A7 0000BFA0 ?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@AEBAPEADPEADDH@Z
        425  1A8 0000BFA0 ?_Ffmt@?$num_put@GV?$ostreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@AEBAPEADPEADDH@Z
        426  1A9 0000BFA0 ?_Ffmt@?$num_put@_WV?$ostreambuf_iterator@_WU?$char_traits@_W@std@@@std@@@std@@AEBAPEADPEADDH@Z

C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29115\bin\Hostx64\x64>undname "?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@AEBAPEADPEADDH@Z"
Microsoft (R) C++ Name Undecorator
Copyright (C) Microsoft Corporation. All rights reserved.

Undecoration of :- "?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@AEBAPEADPEADDH@Z"
is :- "private: char * __ptr64 __cdecl std::num_put<char,class std::ostreambuf_iterator<char,struct std::char_traits<char> > >::_Ffmt(char * __ptr64,char,int)const __ptr64"
```

(We have machinery to validate that the DLL export surface is unchanged, but that has not yet been ported from the MSVC-internal MSBuild system to the GitHub CMake/Ninja build system.)

To preserve bincompat, I think you'll have to drop the `_Has_precision` optimization here, so the format string will always contain `".*"` and you'll need to pass `_Precision` calculated above.

Reply:
> so the format string will always contain ".*" and you'll need to pass _Precision calculated above.

I think the format string should not contain ".*" (the standard says no precision). There's no guarantee that "%.13a" prints exactly the same string as "%a".